src/run_program.rs

```diff
@@ -277,7 +267,7 @@ impl<'a, Handler: OperatorHandler> RunProgramContext<'a, Handler> {
             return err(operator, "internal error");
         }
         let op_atom = self.allocator.atom(operator);
-        if op_atom == self.apply_kw {
+        if op_atom == self.dialect.apply_kw() {
```

Comment:
Might be worth seeing whether this compiles to a function call or gets inlined. A function call for each `q` check would be quite a bit more expensive.

Reply:
it's a static call returning a constant value, so this certainly *could* be cheaper than comparing with an actual `Vec`